app/models/orchestration_template.rb

```diff
@@ -20,6 +20,25 @@ class OrchestrationTemplate < ApplicationRecord
   attr_accessor :remote_proxy
   alias remote_proxy? remote_proxy
 
+  DEPRECATED_TYPES = {
+    'OrchestrationTemplateCfn'   => 'ManageIQ::Providers::Amazon::CloudManager::OrchestrationTemplate',
+    'OrchestrationTemplateHot'   => 'ManageIQ::Providers::Openstack::CloudManager::OrchestrationTemplate',
+    'OrchestrationTemplateVnfd'  => 'ManageIQ::Providers::Openstack::CloudManager::VnfdTemplate',
+    'OrchestrationTemplateAzure' => 'ManageIQ::Providers::Azure::CloudManager::OrchestrationTemplate',
+  }.freeze
+
+  # Temporarily maintain backward compatibility for API users. Should be removed later.
+  def self.new(*args, &block)
+    if (h = args.first).kind_of?(Hash)
+      old_type = h[:type] || h['type']
+      if (new_type = DEPRECATED_TYPES[old_type]).present?
+        _log.warn("#{old_type} has been renamed to #{new_type}")
```

Comment:
@bdunne Is there a method from `DeprecationMixin` we should/could use here?  I thought we had something for this but I'm not seeing it.

Reply:
```
Vmdb::Deprecation.deprecate_methods(class, :old_method => :new_method)
```

Reply:
We are not deprecating a method, but a used valid attribute value.